src/Marten.Testing/Harness/IntegrationContext.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using Marten.Testing.Examples;
+using Xunit;
+
+namespace Marten.Testing.Harness
+{
+    [Collection("integration")]
+    public class IntegrationContext: StoreContext<DefaultStoreFixture>
+    {
+        private IDocumentSession _session;
+        private DocumentStore _store;
+
+        public IntegrationContext(DefaultStoreFixture fixture) : base(fixture)
+        {
+
+        }
+
+        /// <summary>
+        /// Customize the store configuration for one off tests.
+        /// The return value is the database schema
+        /// </summary>
+        /// <param name="configure"></param>
+        /// <returns></returns>
+        protected string StoreOptions(Action<StoreOptions> configure)
+        {
+            _session?.Dispose();
+            _session = null;
+
+            var options = new StoreOptions();
+            options.Connection(ConnectionSource.ConnectionString);
+
+            // Can be overridden
+            options.AutoCreateSchemaObjects = AutoCreate.All;
+            options.NameDataLength = 100;
+            options.DatabaseSchemaName = "special";
+
+            configure(options);
+
+            _store = new DocumentStore(options);
+
+            _store.Advanced.Clean.CompletelyRemoveAll();
+
+            return options.DatabaseSchemaName;
+        }
+
+        private bool _hasBuiltStore = false;
+
+        protected override DocumentStore theStore
+        {
+            get
+            {
+                if (_store != null) return _store;
+
+                if (!_hasBuiltStore)
+                {
+                    base.theStore.Advanced.Clean.DeleteAllDocuments();
+                    _hasBuiltStore = true;
+                }
+
+                return base.theStore;
+            }
+        }
+
+        protected virtual IDocumentSession theSession
```

Comment:
**FIX:** I think that this should be removed as it hides the same property from the base class. We should also consider removing [_session](https://github.com/JasperFx/marten/pull/1463/files#diff-8f22a8190d9167240cb5140d183b18fbR10) from this class and using it from the base.

Reply:
Done.